### ISSUE TITLE
Ruby 2.7.0 causing issues with Nokogiri and Jekyll

### ISSUE THREAD

**Issue by jkillingsworth** [msys2/MINGW-packages#6222]
The recent Ruby update from version 2.6.5 to version 2.7.0 might cause problems for anyone who uses the Nokogiri and/or Jekyll gems. These two gems are not quite ready for Ruby 2.7.0. I am opening this issue to share notes and offer workarounds for anyone else who might get bitten by this.

**Nokogiri.** The latest release version is 1.10.8, which does not work with Ruby 2.7.0. Support for Ruby 2.7.0 is planned for the 1.11.0 release. See [issue #1961](https://github.com/sparklemotion/nokogiri/issues/1961) on in Nokogiri repo for more details. There is a release candidate for version 1.11.0 available. You can place the following in your Gemfile to force it to use the pre-release version:

    gem "nokogiri", ">= 1.11.0.rc1"

**Jekyll.** The latest version is 4.0. The latest version of Jekyll seems to be working fine with Ruby 2.7.0, but it spits out a bunch of warning messages regarding the use of deprecated language features. A fix has already been implement (commit [389eb88](https://github.com/jekyll/jekyll/commit/389eb88c20aa205ad0c412d83b3e60fff577113b)) and is slated for inclusion in the 4.1 release. See [issue #7947](https://github.com/jekyll/jekyll/issues/7947) and [pull request #7948](https://github.com/jekyll/jekyll/pull/7948) in the Jekyll repo for more details. You can place the following in your Gemfile if you want to get rid of the warnings and don't mind being on the bleeding edge:

    gem "jekyll", github: "jekyll/jekyll", branch: "389eb88"

These are temporary workarounds. These problems should go away once Nokogiri 1.11.0 and Jekyll 4.1 are released.

**Comment by jkillingsworth** [comment]
If you are using Jekyll with a Gemfile and the latest version of Bundler (2.1.4), you might run into errors after upgrading to Ruby 2.7.0 if you run the Jekyll local server with this command:

    jekyll serve

This is the error message I get:

```
Warning: the running version of Bundler (2.1.2) is older than the version that created the lockfile (2.1.4). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.1.4`.
C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/runtime.rb:312:in `check_for_activated_spec!': You have already activated mercenary 0.3.6, but your Gemfile requires mercenary 0.4.0. Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/runtime.rb:31:in `block in setup'
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/spec_set.rb:147:in `each'
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/spec_set.rb:147:in `each'
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/runtime.rb:26:in `map'
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler/runtime.rb:26:in `setup'
        from C:/opt/msys64/mingw64/lib/ruby/2.7.0/bundler.rb:149:in `setup'
        from C:/opt/msys64/mingw64/lib/ruby/gems/2.7.0/gems/jekyll-4.0.0/lib/jekyll/plugin_manager.rb:52:in `require_from_bundler'
        from C:/opt/msys64/mingw64/lib/ruby/gems/2.7.0/gems/jekyll-4.0.0/exe/jekyll:11:in `<top (required)>'
        from C:/opt/msys64/mingw64/bin/jekyll:23:in `load'
        from C:/opt/msys64/mingw64/bin/jekyll:23:in `<main>'
```

This happens when I run `jekyll serve` after using `bundle update` to apply the Gemfile changes mentioned in my previous comment. The [guidance from Jekyll](https://jekyllrb.com/docs/step-by-step/10-deployment/#gemfile) and Ruby experts is that you should run commands like this with the `bundle exec` prefix to insure you are using the correct gem versions for your project:

    bundle exec jekyll serve

This runs the local server with no error. However, it spawns an extra Ruby process when you do it this way. If you press `ctrl-c` to stop the server, the extra process remains running. The local server is still serving up pages even after you think you stopped it. Every time you stop and restart the server, the system accumulates additional instances of this errant process. You have to kill them with the task manager.

I don't think this is the expected behavior and I don't think it is specific to the latest version of Ruby. I think the problem lies somewhere between Bash and Bundler. I have not been able to reproduce this behavior outside of the MSYS2 environment. I get correct behavior when running this command from the Windows command prompt with a standalone installation of Ruby installed, hence my suspicion that is has something to do with Bash.

There is a workaround which I think is a better solution than using the `bundle exec` prefix. You can use [binstubs](https://github.com/rbenv/rbenv/wiki/Understanding-binstubs). Bundler lets you generate project-specific binstubs like this:

    bundle binstubs jekyll --path=bin

Run the command above in the same directory as your Gemfile. You can then run the Jekyll local server using the generated binstub:

    bin/jekyll serve

This will run the command with the correct version of all dependencies according to your Gemfile without using the `bundle exec` prefix. Doing it this way does not generate any stray processes.

**Comment by jkillingsworth** [comment]
The Jekyll issue has now been fixed as of version 4.0.1.

**Comment by jkillingsworth** [comment]
The Nokogiri issue has now been fixed as of version 1.11.0.
